File: src/main/java/org/aldo/beautycenter/service/interfaces/RoomService.java
package org.aldo.beautycenter.service.interfaces;

import org.aldo.beautycenter.data.dto.create.CreateRoomDto;
import org.aldo.beautycenter.data.dto.responses.RoomDto;
import org.aldo.beautycenter.data.dto.updates.UpdateRoomDto;

import java.util.List;

public interface RoomService {
    List<RoomDto> getAllRooms();

    RoomDto createRoom(CreateRoomDto createRoomDto);

    void updateRoom(UpdateRoomDto updateRoomDto);

    void deleteRoom(String roomId);
}

File: lib/api/RoomService.dart
import 'package:dio/dio.dart';

import '../model/UpdateRoomDto.dart';
import 'ApiService.dart';

class RoomService {
  final Dio _dio = ApiService.instance.dio;
  final String _path = "/room";

  Future<Response?> getAllRooms() async {
    try {
      return await _dio.get("$_path/all");
    } on DioException catch(e){
      return e.response;
    }
  }

  Future<Response?> updateRoom({required UpdateRoomDto updateRoomDto}) async {
    try {
      return await _dio.patch(_path, data: updateRoomDto.toJson());
    } on DioException catch(e){
      return e.response;
    }
  }

  Future<Response?> createRoom({
    required String name,
    required List<String> services
  }) async {
    try {
      FormData formData = FormData.fromMap({
        "name": name,
        "services": services,
      });

      return await _dio.post(
          _path,
          data: formData,
          options: Options(headers: {"Content-Type": "multipart/form-data"})
      );
    } on DioException catch(e){
      return e.response;
    }
  }


  Future<Response?> deleteRoom({required String roomId}) async {
    try {
      return await _dio.delete(
          _path,
          queryParameters: {"roomId" : roomId}
      );
    } on DioException catch(e){
      return e.response;
    }
  }
}